src/test/java/com/mongodb/kafka/connect/sink/StartedMongoSinkTaskTest.java

```diff
@@ -240,18 +256,40 @@ static boolean match(
     }
   }
 
-  private static final class InMemoryErrorReporter
-      implements Consumer<MongoProcessedSinkRecordData> {
-    private final List<MongoProcessedSinkRecordData> reported = new ArrayList<>();
+  private static final class InMemoryErrorReporter implements BiConsumer<SinkRecord, Exception> {
+    private final List<ReportedData> reported = new ArrayList<>();
 
     @Override
-    public void accept(final MongoProcessedSinkRecordData mongoProcessedSinkRecordData) {
-      reported.add(mongoProcessedSinkRecordData);
+    public void accept(final SinkRecord record, final Exception e) {
```

Comment:
`MongoProcessedSinkRecordData` already supports a record and an exception - could / should it be used?

Reply:
`MongoProcessedSinkRecordData.exception` represents an exception that happens during [post-processing](https://docs.mongodb.com/kafka-connector/current/sink-connector/fundamentals/post-processors/). Refactoring the code in a way that uses this field to express exceptions that happen when writing to MongoDB is not only inconvenient (try to imagine how you would change the `StartedMongoSinkTask.bulkWriteBatch` to pass bulk write exceptions via `MongoProcessedSinkRecordData.exception`), but also violates the semantics of the `MongoProcessedSinkRecordData.exception`.

Reply:
I can live with that - although processed is an open term processed by what? the connector, mongodb?  

Had this work been done at the same time I'd probably have gone for `PostProcessedSinkRecordData` as a more explicit name for the class.

Reply:
Ross, do you want me to change the class name from `MongoProcessedSinkRecordData` to `PostProcessedSinkRecordData` in this PR?